### ISSUE TITLE
No notifications from open chat when Chrome is in background

### ISSUE THREAD

**Issue by RTechSn** [nextcloud/notifications#1946]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce

1. Have a chat with another account open.
2. Have no notifications from other apps
3. Minimize or obstruct your browser window
4. Send a message from another account

### Expected behaviour
Notifications should appear

### Actual behaviour
Notifications appear after unminimizing the window, or not appear at all

## Talk app

**Talk app version:** 19.0.0

**Custom Signaling server configured:** yes/no and version (see  Talk administration settings: `/index.php/index.php/settings/admin/talk#signaling_server`)

**Custom TURN server configured:** Yes

**Custom STUN server configured:** Yes

## Browser

**Microphone available:** yes

**Camera available:** no

**Operating system:** Windows

**Browser name:** Chrome

**Browser version:** 123

**Comment by ShGKme** [comment]
We are checking only active/inactive state. We can also check for frozen/ discarded states for suspended tabs.

I'd say that a proper solution is to move notifications completely out from any DOM rendering, best to a worker to not be a part of the page that can be frozen.

**Comment by nickvergessen** [comment]
While yes, let's first do the quick fix of moving it to the root component

**Comment by ShGKme** [comment]
For the reference:

- Page Lifecycle API: https://developer.chrome.com/docs/web-platform/page-lifecycle-api
- Lifecycle diagram: https://developer.chrome.com/docs/web-platform/page-lifecycle-api/image/page-lifecycle-api-state.svg
- See and change page state: `chrome://discards`

---

In a frozen state, DOM is not updated, but background tasks are still running.

In a discarded state, no new task from the task queue will be executed.

So, to support discarded state or true push notifications, we would need to use workers what most of our libraries do not support.

**Comment by RTechSn** [comment]
This particular issue seems to be fixed in the last update (29.0.3). 
Should this issue be closed, or should it remain open, until the proper solution with the support of discarded state be implemented?

**Comment by Antreesy** [comment]
> should it remain open, until the proper solution with the support of discarded state be implemented?

Yes, we wanted to keep track it and look into web workers usage in the future for handling notifications. You may unsubscribe from receiving a notifications from this issue to not suddenly be bothered one day. Thanks again for the help with investigation!

**Comment by MohammedNoureldin** [comment]
Hi

I have issues generally in Nextcloud notifications. In both Desktop PWA and in browser. The notifications do not arrive and are not shown when the Windows is minimized (PWA and NC webpage). As soon as I open the Window, the notifications are shown (not always quickly, but at least they arrive).

The notifications are in general too slow.

Is it related to this issue and planned for 29.0.4, or is it not related?

**Comment by nickvergessen** [comment]
> I am using 29.0.0

Then update please. Mostlikely it's fixed in 29.0.2 via https://github.com/nextcloud/notifications/pull/1948

**Comment by MohammedNoureldin** [comment]
Thanks a lot @nickvergessen! I confirm that it works now with 29.0.3.

Nevertheless, the notification on the desktop (Web or PWA) is slow, it takes about 10-20 seconds for the notification to arrive and get shown in Windows. Is there anything we can do for it?

**Comment by nickvergessen** [comment]
> Is there anything we can do for it?

Yeah we do 30 second polling. So can be between 0 and 29 seconds.
You can install this additional tool that will create a websocket connection with the clients and browser so that notifications can be pushed live: https://github.com/nextcloud/notify_push
